data_safe_haven/commands/context.py

```diff
@@ -160,22 +173,39 @@ def remove(
     key: Annotated[str, typer.Argument(help="Name of the context to remove.")],
 ) -> None:
     """Removes a context."""
-    settings = ContextSettings.from_file()
+    try:
+        settings = ContextSettings.from_file()
+    except DataSafeHavenConfigError as exc:
+        print("No context configuration file, so no contexts to remove.")
+        raise typer.Exit(code=1) from exc
     settings.remove(key)
     settings.write()
 
 
 @context_command_group.command()
 def create() -> None:
     """Create Data Safe Haven context infrastructure."""
-    context = ContextSettings.from_file().assert_context()
+    try:
+        context = ContextSettings.from_file().assert_context()
+    except DataSafeHavenConfigError as exc:
+        print(
+            "No context configuration file. Run `dsh context add` before creating infrastructure."
+        )
+        raise typer.Exit(code=1) from exc
+
```

Comment:
This isn't quite correct, `ContextSettings.assert_context()` can also raise `DataSafeHavenConfigError` if no context is selected.